tests/func/test_pipeline.py

```diff
@@ -98,6 +99,28 @@ def test_dot_commands(self):
         self.assertEqual(ret, 0)
 
 
+def test_disconnected_stage(tmp_dir, dvc, caplog):
+    tmp_dir.dvc_gen({"base": "base"})
+
+    dvc.add("base")
+    dvc.run(deps=["base"], outs=["derived1"], cmd="echo derived1 > derived1")
+    dvc.run(deps=["base"], outs=["derived2"], cmd="echo derived2 > derived2")
+    final_stage = dvc.run(
+        deps=["derived1"], outs=["final"], cmd="echo final > final"
+    )
+
+    args = ["pipeline", "show", "--outs", "--dot", final_stage.path]
+    command = CmdPipelineShow(args)
+    # Need to test __build_graph directly
+    nodes, edges, is_tree = command._CmdPipelineShow__build_graph(
+        final_stage.path, commands=False, outs=True
+    )
+
+    assert set(nodes) == set(["final", "derived1", "base"])
```

Comment:
no need for converting list to set, `set(["final", "derived1", "base"])` == `{"final", "derived1", "base"}`